test/config/config.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+const test = require('tape')
+
+const { CONFIG_DEFINITIONS } = require('./fixtures/config-schema')
+
+const Agent = require('../../lib/agent')
+const { NoopTransport } = require('../../lib/noop-transport')
+const MockLogger = require('./_mock_logger')
+
+// Options to pass to `agent.start()` to turn off some default agent behavior
+// that is unhelpful for these tests.
+const agentOpts = {
+  centralConfig: false,
+  captureExceptions: false,
+  metricsInterval: '0s',
+  cloudProvider: 'none',
+  logLevel: 'warn'
+}
+const agentOptsNoTransportNoLogger = Object.assign(
+  {},
+  agentOpts,
+  {
+    logger: new MockLogger(),
+    transport: function createNoopTransport () {
+      // Avoid accidentally trying to send data to an APM server.
+      return new NoopTransport()
+    }
+  }
+)
+
+function hasType (def, type) {
+  return def.types && def.types.indexOf(type) !== -1
+}
+
+function testEnvVar (t, option, value1, value2) {
+  const existingEnvValue = process.env[option.envVar]
+  const agent = new Agent()
+  const opts = {}
+  let value = value1
+
+  if (value2) {
+    opts[option.name] = value1
+    value = value2
+  }
+
+  process.env[option.envVar] = value.toString()
+  agent.start(agentOptsNoTransportNoLogger)
+  t.strictEqual(agent._conf[option.name], value)
+
+  process.env[option.envVar] = existingEnvValue
+  agent.destroy()
+  t.end()
```

Comment:
nit: My preference would be that utility test functions do *not* call the `t.end()` themselves. While I know this means that the tests below need to write `t.end()` more times, it *does* mean that it is a bit easier to read a given test block:

```js
test(..., t => {
    ...
})
```

and see if it explicitly calls `t.end()`.